src/providers/mod.rs

```diff
@@ -64,6 +67,23 @@ fn create_file(filename: &str) -> Result<File> {
     File::create(file_path).chain_err(|| format!("failed to create file {:?}", file_path))
 }
 
+fn write_structured_journald_entry(
+    log: logging::Priority,
+    msg: &str,
+    name: &str,
+    path: &str,
+) -> Result<()> {
+    let map = maplit::hashmap! {
+        "AFTERBURN_USER_NAME" => name.as_ref(),
+        "AFTERBURN_PATH" => path.as_ref(),
+        "MESSAGE_ID" => AFTERBURN_SSH_AUTHORIZED_KEYS_MESSAGEID,
+    };
+
+    if let Err(e) = logging::journal_send(log, msg, map.iter()) {
+        warn!("failed to send information to journald: {}", e);
```

Comment:
Of course, with this message we'll also try to send it to the journal...something will have had to go catastrophically wrong for it to fail.